libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -739,14 +669,47 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
         *pOperationReference = pSubscriptionOperation;
     }
 
-    /* Schedule the subscription operation for network transmission. */
-    status = _IotMqtt_ScheduleOperation( pSubscriptionOperation,
-                                         _IotMqtt_ProcessSend,
-                                         0 );
+    if( pSubscriptionOperation->u.operation.status == IOT_MQTT_STATUS_PENDING )
```

Comment:
Why is this check required?

Reply:
This was to check whether the operation is created successfully or not.As we are asserting already, we can drop this check.